src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -120,8 +120,9 @@ public JsonWebToken(string header, string payload)
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14302, payload), ex));
             }
 
-            _encodedHeader = header;
-            _encodedPayload = payload;
+            _encodedHeader = Base64UrlEncoder.Encode(header);
```

Comment:
This is different than the logic in the EncodedHeader where `encodedHeader = EncodedToken.Substring(0, Dot1)`

        public string EncodedHeader
        {
            get
            {
                // TODO - need to account for JWE
                if (_encodedHeader == null)
                {
                    if (EncodedToken != null)
                        _encodedHeader = EncodedToken.Substring(0, Dot1);
                    else
                        _encodedHeader = string.Empty;
                }

                return _encodedHeader;
            }
        }